util/circular-chk.py

```diff
@@ -0,0 +1,51 @@
+import glob
+import re
+from sys import exit, argv
+from pathlib import Path
+
+strict = False
+if len(argv) > 1 and argv[1] == '--strict' :
+    strict = True
+
+regexExclude = r"keymaps\/(.*)via\/"
+reg = re.compile(regexExclude)
+filesToscan = Path('keyboards').glob('**/rules.mk')
+illegalSRC = list()
+
+for f in filesToscan:
+    p = str(f).replace('\\','/')
+    basepath = p.replace('keyboards/','')
+    basepath_arr = basepath.split('/')
+
+    # Skip 'via' file
+    if reg.search(p):
```

Comment:
```suggestion
    if file.match('**/*via/rules.mk')
```

This should also be at the top of the loop so we don't do unnecessary processing.

Reply:
What if there's no `rules.mk` file?

Reply:
Then the directory won't be in the list anyway, since your glob is `**/rules.mk`.

Reply:
My bad forgot this part.